### PR TITLE
Add Collections, Mnemonic

### PR DESCRIPTION
**Pull request by SeeuSim**
# [DEPLOYMENT LINK](https://nfinsight-staging.vercel.app)

- Add Collections schema and migration
- Add env variable validation
- Modified existing env variable usage to validate environment variables
- Added Mnemonic API env variable
- Added Mnemonic API endpoint types
- Added Mnemonic Utility File under `./src/utils`
- Added Mnemonic Collections Metadata endpoint

### CODE REVIEW

File: src/pages/collection/[address].tsx [hunk 1]
```diff
@@ -145,17 +190,128 @@ export default function Collection() {
...
+          })}
+        </select>
+        <select 
+          className="rounded-md p-2 bg-slate-100 hover:bg-slate-200 dark:bg-slate-700 dark:text-slate-300 dark:hover:bg-slate-600 shadow-md"
+          value={records} 
+          onChange={(e: any & {target: {value: number}}) => setRecords(e.target.value)}
+          >
+          <option key={7} value={7}>7d</option>
+          <option key={30} value={30}>30d</option>
+          <option key={365} value={365}>1y</option>
```
Comment (JamesLiuZX) [inline src/pages/collection/[address].tsx:211]: Key is default days. Could add support for hourly and 4hourly charts

File: src/pages/collection/[address].tsx [hunk 2]
```diff
@@ -145,17 +190,128 @@ export default function Collection() {
...
   return (
-    <div className="py-4 grid grid-cols-1 lg:grid-cols-2 gap-x-4 gap-y-4 grid-rows-2 lg:grid-rows-1 h-full">
+    <div className="grid grid-cols-1 lg:grid-cols-2 gap-x-4 gap-y-4 grid-rows-2 lg:grid-rows-1">
       <Head>
-        <title>{`Collection ${address} | DogeTTM`}</title>
+        <title>{`Collection ${collection.name || address} | NFinsighT`}</title>
       </Head>
-      {pane}
-      {graphPane}
+      <Suspense fallback={<div className="text-9xl m-10 text-white">Loading</div>}>
```
Comment (JamesLiuZX) [inline src/pages/collection/[address].tsx:226]: Would be cool if we could add a spinner component and center it with the 'loading' message for our fallbacks across the app. Not very important now but will be more relevant as the DB gets larger, and some clients being slower.

File: src/pages/collection/[address].tsx [hunk 3]
```diff
@@ -145,17 +190,128 @@ export default function Collection() {
...
+
+      <div className="px-4">
+        <LineGraph width={800} height={400} yCallback={yAxisCallback} graphLabel={selector} labels={graphLabels} dataPts={dataPts}/> 
+      </div>
     </div>
 
 
   return (
-    <div className="py-4 grid grid-cols-1 lg:grid-cols-2 gap-x-4 gap-y-4 grid-rows-2 lg:grid-rows-1 h-full">
+    <div className="grid grid-cols-1 lg:grid-cols-2 gap-x-4 gap-y-4 grid-rows-2 lg:grid-rows-1">
```
Comment (JamesLiuZX) [inline src/pages/collection/[address].tsx:222]: We could center the graph further down so it won't look so empty on the bottom margin.

File: src/components/graphs/lineGraph.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  labels: string[],
+  dataPts: number[]
+}> = ({ dataPts, graphLabel, labels, yCallback, width, height }) => {
+  const options = {
+    responsive: true,
+    plugins: {
+      legend: {
+        labels: {
+          font: {
+            size: 14,
```
Comment (JamesLiuZX) [inline src/components/graphs/lineGraph.tsx:39]: size: 18/20

File: src/components/graphs/lineGraph.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            family: "system-ui"
+          },
+          color: "#64748b"
```
Comment (JamesLiuZX) [inline src/components/graphs/lineGraph.tsx:42]: Add `color: "#FFFFFF"` to make title more visible

File: src/components/graphs/lineGraph.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    },
+    scales: {
+      y: {
+        type: 'linear' as const,
+        display: true,
+        position: 'left' as const,
+        min: 0,
+        ticks: {
+          callback: yCallback,
+          color: "#64748b"
```
Comment (JamesLiuZX) [inline src/components/graphs/lineGraph.tsx:68]: Same for the rest of the label colors

**Review by JamesLiuZX — DISMISSED** [review]
Great PR. Nothing structurally or logically breaking. Mostly on the UI side of [address].tsx and the graph.

File: src/utils/ethereum/price.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import type { Decimal } from "@prisma/client/runtime";
+
+export const formatVal = (v?: string | Decimal) => {
+  if (!v) return "0"
+  const actual = new Number(v);
+  return actual > 1e9
+    ? `${actual.toExponential(3).toLocaleUpperCase("en-US")}`
```
[the remark this answers is not in the document]
Comment (SeeuSim) [reply src/utils/ethereum/price.ts:7]: This is specific formatting for display, not specific to any region

File: src/pages/account/settings/(settings-components)/oAuthSettingsModal.tsx [hunk 8]
```diff
@@ -60,11 +59,11 @@ const ModalUI: React.FC<{
 }
```
[the remark this answers is not in the document]
Comment (SeeuSim) [reply src/pages/account/settings/(settings-components)/oAuthSettingsModal.tsx:60]: DeleteProfileCallback is a WIP feature to allow users to unlink OAuth accounts, to be added in the future

[the remark this answers is not in the document]
Comment (SeeuSim) [reply src/pages/account/settings/(settings-components)/oAuthSettingsModal.tsx:60]: id will be used to retrieve the profile and delete it

File: src/pages/collection/[address].tsx (hunk 1 above)
Comment (SeeuSim) [reply src/pages/collection/[address].tsx:211]: WIP feature to be added. As of now database does not have capacity or speed to support hourly checkpoints

File: src/pages/collection/[address].tsx (hunk 2 above)
Comment (SeeuSim) [reply src/pages/collection/[address].tsx:226]: Suspense fallback does not work in Next12, WIP transition to client side rendering with loading fallback

File: src/components/graphs/lineGraph.tsx (hunk 4 above)
Comment (SeeuSim) [reply src/components/graphs/lineGraph.tsx:39]: fixed

File: src/components/graphs/lineGraph.tsx (hunk 5 above)
Comment (SeeuSim) [reply src/components/graphs/lineGraph.tsx:42]: Will not render properly in light mode. As of now, this is a compromise as the dynamic props do not re-render the graph based off changes in the page.

File: src/pages/collection/[address].tsx (hunk 3 above)
Comment (SeeuSim) [reply src/pages/collection/[address].tsx]: fixed

**Pull request merged**